pkg/api/v1/types.go

```diff
@@ -3404,6 +3404,15 @@ type PodExecOptions struct {
 	Command []string `json:"command" protobuf:"bytes,6,rep,name=command"`
 }
 
+// PodPortForwardOptions is the query options to a Pod's port forward call.
+type PodPortForwardOptions struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// List of ports to forward
```

Comment:
SPDY expects the port information as a HTTP Header.
This is where backward compatibility and the capabilities of the transports do not translate well. SPDY which was already there passed the Ports as a HEADER in the request(s). SPDY allows multiple requests.
Websockets does not allow HEADERs to be set so it had to be ports in a query parameter.

To my knowledge there is no way to document the HEADER as part of the API and that is what is there today.